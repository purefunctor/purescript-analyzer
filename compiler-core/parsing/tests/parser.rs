use test_each_file::test_each_file;

test_each_file! { in "./compiler-core/parsing/tests/parser" => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, errors) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    insta::assert_debug_snapshot!((node, errors));
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as lossless => |content: &str| {
    let lexed = lexing::lex(content);
    let tokens = lexing::layout(&lexed);
    let (parsed, _) = parsing::parse(&lexed, &tokens);
    let node = parsed.syntax_node();
    assert_eq!(node.to_string(), content);
}}

test_each_file! { in "./compiler-core/parsing/tests/parser" as stability => |content: &str| {
    let lexed = lexing::lex(content);
    for index in 0..lexed.len() - 1 {
        let partial = lexed.text_in_range(0..index + 1);
        let lexed = lexing::lex(partial);
        let tokens = lexing::layout(&lexed);
        let (parsed, _) = parsing::parse(&lexed, &tokens);
        let node = parsed.syntax_node();
        assert_eq!(node.to_string(), partial);
    }
}}
