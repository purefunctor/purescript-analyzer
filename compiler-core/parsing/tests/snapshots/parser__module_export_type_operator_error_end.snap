---
source: compiler-core/parsing/tests/parser.rs
expression: "(node, errors)"
---
(
    Module@0..53
      ModuleHeader@0..52
        MODULE@0..6 "module"
        ModuleName@6..39
          Annotation@6..7
            TEXT@6..7 " "
          UPPER@7..39 "ModuleExportTypeOpera ..."
        ExportList@39..46
          Annotation@39..40
            TEXT@39..40 " "
          LEFT_PARENTHESIS@40..41 "("
          ExportTypeOperator@41..45
            TYPE@41..45 "type"
            ERROR@45..45 ""
          RIGHT_PARENTHESIS@45..46 ")"
        Annotation@46..47
          TEXT@46..47 " "
        WHERE@47..52 "where"
      LAYOUT_START@52..52 ""
      LAYOUT_END@52..52 ""
      Annotation@52..53
        TEXT@52..53 "\n"
      END_OF_FILE@53..53 ""
    ,
    [
        ParseError {
            offset: 45,
            position: Position {
                line: 1,
                column: 46,
            },
            message: "Expected OPERATOR_NAME",
        },
    ],
)
